viz/cmd/tap.go

```diff
@@ -428,15 +460,60 @@ func renderTapEvent(event *tapPb.TapEvent) string {
 }
 
 // renderTapEventJSON renders a Public API TapEvent to a string in JSON format.
-func renderTapEventJSON(event *tapPb.TapEvent) string {
+func renderTapEventJSON(event *tapPb.TapEvent, opts ...renderOptions) string {
+	filter := &renderFilter{}
+	for _, opt := range opts {
+		opt(filter)
+	}
 	m := mapPublicToDisplayTapEvent(event)
+	if filter.JsonPath != "" {
+		filteredJson, err := getJsonFilteredByJPath(m, filter.JsonPath)
+		if err != nil {
+			return err.Error()
+		}
+		return filteredJson[0]
+	}
 	e, err := json.MarshalIndent(m, "", "  ")
 	if err != nil {
 		return fmt.Sprintf("{\"error marshalling JSON\": \"%s\"}", err)
 	}
 	return string(e)
 }
 
+func getJsonFilteredByJPath(event interface{}, jsonPath string) ([]string, error) {
+	fields, err := get.RelaxedJSONPathExpression(jsonPath)
+	if err != nil {
+		return []string{}, err
+	}
+
+	j := jsonpath.New("EventParser")
+	if err := j.Parse(fields); err != nil {
+		return []string{}, err
+	}
+
+	results, err := j.FindResults(event)
+	if err != nil {
+		return []string{}, err
```

Comment:
Looking at the client-go implementation, it seems some errors are not rendered in JSON format? e.g. see [this link](https://github.com/kubernetes/client-go/blob/master/util/jsonpath/jsonpath.go#L82-L83). Can we confirm whether all errors here are JSON encoded strings?